### PR TITLE
Add chained-comparison and literal-macro-adjacency diagnostics (#266, #267)

### LINKED ISSUE #266: Detect suspicious chained comparisons in Stata expressions
## Summary

Sight should warn on suspicious chained comparisons such as:

```stata
if (a != b != c) {
    display "unexpected"
}
```

This came up from a missing logical connector in a real expression:

```stata
// missing `&` before `c'
if (`a' == 1 & `b' != 1`c' != 1 ) {

// corrected
if (`a' == 1 & `b' != 1 & `c' != 1 ) {
```

## Current behavior

Sight does not currently emit a diagnostic for a comparison chain. In the missing-`&` example, the source token stream includes a comparison, then an operand-like token, then another comparison, but no rule reports it once the macros are defined.

## Expected behavior

Emit a warning/hint for comparison chains like:

```stata
a != b != c
a == b == c
a < b < c
a <= b <= c
```

The diagnostic should explain that Stata does not have mathematical chained-comparison semantics and that the expression is often a missing logical operator or missing parentheses.

Suggested wording:

> Chained comparison `a != b != c` is suspicious. Use an explicit logical operator or parentheses to make the intended evaluation clear.

## Implementation notes

This likely fits alongside the existing token-stream diagnostics in `src/providers/`, similar to `MixedLogicalOperatorAnalyzer` and `OperatorSequenceAnalyzer`.

The rule should be careful about expression boundaries and embedded contexts. It should not produce duplicate diagnostics for already-invalid operator sequences.

### LINKED ISSUE #267: Warn on suspicious literal-macro adjacency in expression contexts
## Summary

Sight should consider warning when a literal is immediately adjacent to a macro reference in an expression context, for example:

```stata
if (a == 1`b') {
    display "unexpected"
}
```

This is valid Stata syntax: macro expansion can intentionally concatenate text. But in an expression like this, it is likely a mistake. For example, if local macro `b` expands to `0`, then `1`b'` becomes `10`, not `1` followed by a separate condition.

## Current behavior

Sight does not emit a diagnostic. The lexer treats this as adjacent tokens, for example:

```text
NUMBER:"1" MACRO_REF_LOCAL:"`b'"
```

That is syntactically valid after Stata macro expansion, so this should not be treated as a parser error.

## Expected behavior

Emit a low-noise warning/hint for suspicious expression-context adjacency such as:

```stata
a == 1`b'
a != "x"`suffix'
a < 10`cutoff'
```

Suggested wording:

> Literal adjacent to macro reference in an expression. Stata will concatenate these during macro expansion; add an operator, whitespace, or parentheses if that was not intended.

## False-positive constraints

Stata code intentionally uses macro adjacency in many places, so the rule should be scoped narrowly. It should avoid warning for common intentional patterns such as:

```stata
generate x`i' = 1
use "data`year'.dta", clear
display "prefix`name'"
```

A good first scope may be expression operands following comparison/logical operators or inside `if`/`while` conditions, rather than all token adjacency.

### PR DESCRIPTION
**Pull request by jbearak**
Closes #266. Closes #267.

## Summary

Adds two new heuristic token-stream diagnostics for Stata, following the existing `OperatorSequenceAnalyzer` / `MixedLogicalOperatorAnalyzer` pattern:

- **`CHAINED_COMPARISON`** (#266, default `warning`): flags two or more comparisons chained without a logical connector — `a != b != c`, `a < b < c`, mixed forms like `a < b > c`. Stata evaluates comparisons left-to-right (`a < b < c` is `(a < b) < c`), so a chain is almost always a missing `&`/`|` or missing parentheses.
- **`LITERAL_MACRO_ADJACENCY`** (#267, default `hint`): flags a number or complete string literal placed directly against a following macro reference when the pair is an operand of a comparison/logical operator — `a == 1\`b'`, `a != "x"\`suffix'`, `1\`b' == a`, `if (1\`b')`. Stata concatenates these during macro expansion (`1\`b'` → `10` if `` `b' ``=`0`), which is usually a mistake.

## Design

- Both analyzers consume a shared **`collect_significant_tokens`** stream (`src/providers/diagnostic-token-stream.ts`) that centralizes trivia / `///`-continuation / comment handling, plus shared operator constants and `is_adjacent`. A shared **`resolve_diagnostic_severity`** (`src/utils/diagnostic-severity.ts`) replaces the per-analyzer severity switches.
- **Chained-comparison** tracks per-paren-depth comparison runs; a comparison counts only after an operand (so adjacent `< <` is left to `OperatorSequenceAnalyzer`), and a whitespace-separated operand→operand transition closes a run at command boundaries (braceless single-line `if`).
- **Literal-macro-adjacency** flags only when a comparison/logical operator governs the pair — scanning past grouping parens and unary prefixes, with call-opener detection (a `(` directly adjacent to a name is a function call, so `strlen(1\`x') > 5` is left alone while `assert (1\`b') == 10` is flagged), and a paren-skip cap so a final function argument (`foo(y, 2\`g') > 0`) is not flagged. Also flags a leading literal in an `if`/`while` condition (`if 1\`b'`).

Fully wired: config severity keys + validation + public schema, `DEFAULT_SETTINGS`, VS Code settings contributions, `docs/diagnostics.md` / `docs/configuration.md`, and the config-schema completeness test. New unit tests cover the issue examples plus an extensive false-positive/false-negative battery (function args, subscripts, `#delimit ;`, string interpolation, `@lsp-ignore`, etc.).

## Review

Reviewed adversarially over several rounds by Codex and `/code-review`; every expression-context correctness finding (braceless-if run-spanning, block/line comments, `///` continuation, parenthesized/signed/bare conditions, function-argument leaks, severity fallback) was fixed and covered by regression tests. Remaining edge cases are inherent to the context-light token-stream approach shared with the existing operator diagnostics (e.g. firing inside text commands like `notes:`) and are tracked in #268 for a future command-context improvement.

`bun run test` (typecheck + full suite) and `bun run lint` pass.

https://claude.ai/code/session_01R7zU2qVeExQB2VB8uAi3XA

## Summary by CodeRabbit

* **New Features**
  * Added two new diagnostics for suspicious chained comparisons and literal/macro adjacency.
  * Added configurable severity settings for both diagnostics, with defaults of warning and hint.

* **Documentation**
  * Updated configuration and diagnostics docs to describe the new checks and example settings.

* **Bug Fixes**
  * Improved diagnostic handling so the new checks respect ignore markers and embedded/non-Stata code contexts.

### CODE REVIEW

File: src/providers/literal-macro-adjacency-diagnostics.ts [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+            const operator_before = is_expression_operator(before.token);
+            const operator_after = is_expression_operator(governor_after);
+            const leads_condition =
+                before.token !== undefined &&
+                before.token.type === 'WORD' &&
+                CONDITION_KEYWORDS.has(before.token.value);
+
+            if (
+                (operator_before || operator_after || leads_condition) &&
+                !my_ignored_lines.has(my_literal.range.start.line)
```
[the remark this answers is not in the document]
Comment (jbearak) [reply src/providers/literal-macro-adjacency-diagnostics.ts:277]: Checked this against the current suppression contract and we are not making this change.

`@lsp-ignore-next` is line-targeted in the implementation: `Analyzer.ignore_next_non_trivia_line` records the line of the next non-trivia token in `ignored_lines`, and each suppressible diagnostic checks whether its own diagnostic line is ignored. In the continued-condition case, the ignored line is the `if a == ///` line; the literal-macro diagnostic is emitted on the following physical continuation line, so reporting it is consistent with the existing model.

`ChainedComparisonAnalyzer` uses range-based suppression because that diagnostic range itself spans the comparison operators. Extending `LITERAL_MACRO_ADJACENCY` to suppress based on a governing expression line would broaden `ignore-next` from next token line to whole logical continued statement, which is a semantic change rather than a bug fix. We should handle any doc ambiguity separately by clarifying the line-based behavior.
